src/react-components/room-settings-dialog.js

```diff
@@ -26,6 +27,31 @@ export default class RoomSettingsDialog extends Component {
     this.props.onClose();
   };
 
+  renderPrivacyDropdown(disabled, onChange) {
+    return (
+      <div className={styles.selectContainer}>
+        <select
+          value={this.state.privacy}
+          onChange={
+            onChange ||
+            (e =>
+              this.setState({
+                privacy: e.target.value
+              }))
+          }
+        >
+          <option value="public">Public (Listed on the homepage)</option>
```

Comment:
two points here:
- need to push these into translation strings
- I didn't load the design, but I would update it so that these are laid out so "Public" and "Private" are top level radios with label color (black or white depending on theme), and then the "Visible on homepage" and "Only those with the link can join" are subtitle font/color underneath each label